persistence/src/main/java/org/cbioportal/genome_nexus/persistence/internal/PdbHeaderRepositoryImpl.java

```diff
@@ -0,0 +1,41 @@
+package org.cbioportal.genome_nexus.persistence.internal;
+
+import org.cbioportal.genome_nexus.model.PdbHeader;
+import org.cbioportal.genome_nexus.model.SimpleCacheEntity;
+import org.cbioportal.genome_nexus.util.PdbHeaderParser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class PdbHeaderRepositoryImpl
+    extends PlainTextMongoRepositoryImpl // this allows us to save the string value into the simple cache collection
+    implements PdbHeaderRepositoryCustom<PdbHeader, String>
+{
+    public static final String COLLECTION = "pdb.header";
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    private PdbHeaderParser parser;
+
+    public PdbHeader findOne(String id)
```

Comment:
We need to implement our own `findOne` method, since we are fetching a `SimpleCacheEntitiy` instance from database, and then manually mapping (parsing) the plain text value to a `PdbHeader` instance

Reply:
Cool, when we need another service that needs plain text we can reuse it